File: frontend/lib/models/diet_prediction/meal_type.dart
enum MealType {
  breakfast(0, 'breakfast'),
  morningSnack(1, 'morning_snack'),
  lunch(2, 'lunch'),
  afternoonSnack(3, 'afternoon_snack'),
  dinner(4, 'dinner'),
  eveningSnac(5, 'evening_snack');

  final int value;
  final String nameStr;

  const MealType(this.value, this.nameStr);

  String toJson() => nameStr;

  static MealType fromJson(String value) {
    return MealType.values.firstWhere(
      (e) => e.nameStr == value,
      orElse: () => throw ArgumentError('Unknown diet type: $value'),
    );
  }

  int toInt() => value;
}

File: backend/diet_prediction/enums/meal_type.py
from enum import Enum


class MealType(Enum):
    BREAKFAST = "breakfast"
    MORNING_SNACK = "morning_snack"
    LUNCH = "lunch"
    AFTERNOON_SNACK = "afternoon_snack"
    DINNER = "dinner"
    EVENING_SNACK = "evening_snack"

    def __str__(self):
        return self.value

    @classmethod
    def _missing_(cls, value):
        for member in cls:
            if member.value == value:
                return member
        return None
